packages/sanity/src/core/store/_legacy/document/listenQuery.ts

```diff
@@ -92,11 +92,23 @@ export function listenQuery(
     return options.transitions.includes(event.transition)
   }
 
+  const doFetch = () => fetch(client, fetchQuery, params, options)
+
   return merge(
     welcome$.pipe(take(1)),
     mutationAndReconnect$.pipe(
       filter(isRelevantEvent),
       throttleTime(options.throttleTime || 1000, asyncScheduler, {leading: true, trailing: true})
     )
-  ).pipe(exhaustMapWithTrailing(() => fetch(client, fetchQuery, params, options)))
+  ).pipe(
+    exhaustMapWithTrailing((event) => {
+      if (event.type === 'mutation' && event.visibility !== 'query') {
+        // Even though the listener request specifies visibility=query, the events are not guaranteed to be delivered with visibility=query
+        // If the event we are responding to arrives with visibility != query, we add a little delay to allow for the updated document to be available for queries
+        // See https://www.sanity.io/docs/listening#visibility-c4786e55c3ff
+        return timer(1200).pipe(mergeMap(doFetch))
```

Comment:
Can you briefly explain why this using a timer here works?
> After given duration, emit numbers in sequence every specified duration
What is making this only do a single refetch instead of one every 1200ms?

Reply:
`timer(duration)` (with a single argument) waits for the `duration`ms, emits `0` and then completes. Sounds like the docs you refer to here is about the timer(duration, interval) overload, which first waits for `duration`, then emits at every `interval`?

Reply:
Ah, yes. Thanks!